source/compatibility.tex

```diff
@@ -1922,6 +1922,34 @@
 
 \rSec2[diff.cpp17.dcl.dcl]{\ref{dcl.dcl}: declarations}
 
+\diffref{dcl.typedef}
+\change Unnamed classes with a typedef name for linkage purposes
+can only contain C-compatible constructs.
+\rationale Necessary for implementability.
+\effect Valid C++ 2017 code may be ill-formed in this International Standard.
+\begin{codeblock}
+typedef struct {
+  void f() {}           // ill-formed; previously well-formed
```

Comment:
FYI - our coding style guidelines say that the starting columns for comments should be multiples of 4, but other examples' comments in [diff.cpp17.dcl.dcl] start on column 25 so I wouldn't change this without changing the others as well.

Reply:
We seem to be measuring differently. This is column 24 in my book.

Reply:
Eek!  You're right!  Vim has been lying to me!  Which means all of *my* comments are off!!!
```
  void f() {}           // ill-formed; previously well-formed
0         1         2
0123456780912345678091234
...
compatibility.tex [+]                                         1932,25        77%
```
The "1932,25" above is what vim thinks is the line and column.

Reply:
I'd guess vim starts counting at 1, but everybody else starts counting at 0. I'm not sure who's right. What would a traditional typewriter do if you hit the "tab" key?

Reply:
Opened #3131 for this.